packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -0,0 +1,1107 @@
+      this._mouseDownCallback
+    );
+
+    element.addEventListener(Events.TOUCH_TAP, this._mouseDownCallback);
+  };
+
+  private _deactivateDraw = (element) => {
+    state.isInteractingWithTool = false;
+
+    element.removeEventListener(Events.KEY_DOWN, this._keyDownCallback);
+    element.removeEventListener(Events.MOUSE_DOWN, this._mouseDownCallback);
+    element.removeEventListener(
+      Events.MOUSE_DOUBLE_CLICK,
+      this._mouseDownCallback
+    );
+
+    element.removeEventListener(Events.TOUCH_TAP, this._mouseDownCallback);
+  };
+
+  /**
+   * it is used to draw the circleROI annotation in each
+   * request animation frame. It calculates the updated cached statistics if
+   * data is invalidated and cache it.
+   *
+   * @param enabledElement - The Cornerstone's enabledElement.
+   * @param svgDrawingHelper - The svgDrawingHelper providing the context for drawing.
+   */
+  renderAnnotation = (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ): boolean => {
+    let renderStatus = false;
+    const { viewport } = enabledElement;
+    const { worldToCanvas } = viewport;
+    const { element } = viewport;
+
+    // If rendering engine has been destroyed while rendering
+    if (!viewport.getRenderingEngine()) {
+      console.warn('Rendering Engine has been destroyed');
+      return renderStatus;
+    }
+
+    let annotations = getAnnotations(this.getToolName(), element);
+
+    if (!annotations?.length) {
+      return renderStatus;
+    }
+
+    annotations = this.filterInteractableAnnotationsForElement(
+      element,
+      annotations
+    );
+
+    if (!annotations?.length) {
+      return renderStatus;
+    }
+
+    const targetId = this.getTargetId(viewport);
+    const newAnnotation = this.editData?.newAnnotation;
+    const styleSpecifier: StyleSpecifier = {
+      toolGroupId: this.toolGroupId,
+      toolName: this.getToolName(),
+      viewportId: enabledElement.viewport.id,
+    };
+
+    for (let i = 0; i < annotations.length; i++) {
+      const annotation = annotations[i] as SplineROIAnnotation;
+      const { annotationUID, data, highlighted } = annotation;
+      const { handles } = data;
+      const { points, activeHandleIndex } = handles;
+
+      styleSpecifier.annotationUID = annotationUID;
+
+      const lineWidth = this.getStyle(
+        'lineWidth',
+        styleSpecifier,
+        annotation
+      ) as number;
+      const lineDash = this.getStyle(
+        'lineDash',
+        styleSpecifier,
+        annotation
+      ) as string;
+      const color = this.getStyle(
+        'color',
+        styleSpecifier,
+        annotation
+      ) as string;
+
+      const canvasCoordinates = points.map((p) =>
+        worldToCanvas(p)
+      ) as Types.Point2[];
+
+      const splineType = annotation.data.spline.type;
+      const splineConfig = this._getSplineConfig(splineType);
+      const spline = this._createOrUpdateAnnotationSpline(element, annotation);
```

Comment:
Is there any scenario where you don't have the spline and you really need to create it? I guess in this line it is more on the update and these lines

```js
 // Update spline resolution in case it has changed
    if (spline.resolution !== splineConfig.resolution) {
      spline.resolution = parseInt(splineConfig.resolution);
      annotation.invalidated = true;
    }

    // Update Cardinal spline scale in case it has changed
    if (
      spline instanceof CardinalSpline &&
      !spline.fixedScale &&
      splineConfig.scale !== undefined &&
      spline.scale !== splineConfig.scale
    ) {
      spline.scale = splineConfig.scale;
      annotation.invalidated = true;
    }

```

maybe we just refactor those and call those here?